### PR TITLE
feat: add command line parser

### PR DESCRIPTION
**Pull request by ajewellamz**
_Issue #, if available:_

_Description of changes:_

_Squash/merge commit message, if applicable:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: StandardLibrary/src/GetOpt.dfy [hunk 1]
```diff
@@ -0,0 +1,435 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+include "../../../../submodules/MaterialProviders/libraries/src/Wrappers.dfy"
+include "../../../../submodules/MaterialProviders/StandardLibrary/src/StandardLibrary.dfy"
```
Comment (seebees) [inline StandardLibrary/src/GetOpt.dfy:5]: I don't think these are the right path?

File: StandardLibrary/src/GetOpt.dfy [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+        else
+          GetMaps(opts[1..], longMap[opt.name := opt.Opt?], shortMap, commandMap)
+  }
+
+  function printFromFunction<T>(x: T): () {
+    ()
+  } by method {
+    print x,"\n";
+    return ();
+  }
```
Comment (seebees) [inline StandardLibrary/src/GetOpt.dfy:271]: I do this too, I was thinking 
```suggestion
  function Print<T>(x: T): Outcome<string> {
    Pass
  } by method {
    print x,"\n";
    return Pass;
  }
```

Then you can
```dafny
:- Print("less typing!");
```

File: StandardLibrary/src/GetOpt.dfy [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+
+  datatype Context = Context (
+    longMap : map<string, bool>,
+    shortMap : map<char, string>,
+    commands : CommandMap,
+    command : string
+  )
+
+  function /*{:tailrecursion}*/ GetOptions2(
```
Comment (seebees) [inline StandardLibrary/src/GetOpt.dfy:290]: What is the diff in the 2?

Comment (ajewellamz) [reply StandardLibrary/src/GetOpt.dfy:290]: GetOptions is the thing the user calls.  GetOptions2 is the recursive thing that implements most of GetOptions.

**Review by seebees — APPROVED** [review]
Love it.
Much more complete that what I hand rolled.
Thanks.

**Pull request merged**